rewrite-gradle-tooling-model/model/src/main/java/org/openrewrite/gradle/marker/GradleProjectBuilder.java

```diff
@@ -98,16 +102,32 @@ static List<MavenRepository> mapRepositories(List<ArtifactRepository> repositori
     }
 
     private static MavenRepository.Builder withAuthentication(MavenArtifactRepository repo, MavenRepository.Builder builder) {
-        try {
-            PasswordCredentials passwordCredentials = repo.getCredentials(PasswordCredentials.class);
+        if (!(repo instanceof AuthenticationSupportedInternal)) {
+            return builder;
+        }
+
+        Credentials credentials = getCredentials((AuthenticationSupportedInternal) repo);
+        if (credentials instanceof PasswordCredentials) {
+            PasswordCredentials passwordCredentials = (PasswordCredentials) credentials;
             Optional.ofNullable(passwordCredentials.getUsername()).ifPresent(builder::username);
             Optional.ofNullable(passwordCredentials.getPassword()).ifPresent(builder::password);
-        } catch (IllegalArgumentException e) {
-            // We're not using password credentials
         }
         return builder;
     }
 
+    @SuppressWarnings("unchecked")
+    private static @Nullable Credentials getCredentials(AuthenticationSupportedInternal authenticatedRepo) {
+        Object configuredCredentials = authenticatedRepo.getConfiguredCredentials();
+        if (configuredCredentials instanceof Property) {
+            // Gradle 6.6+ returns a Property<Credentials> instance
+            return ((Property<Credentials>) configuredCredentials).getOrNull();
+        } else if (configuredCredentials instanceof Credentials) {
+            // Gradle < 6.6 returns a Credentials instance
+            return (Credentials) configuredCredentials;
+        }
+        return null;
+    }
```

Comment:
We can inline this if we'd prefer, it just seemed like something simple to externalize and keep the comments closely related/clean.

Reply:
Fine indeed as it is; I like the early returns that would otherwise have to be assignments to the same variable.